consensus/core/src/tests/base_committer_declarative_tests.rs

```diff
@@ -0,0 +1,367 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::panic;
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+use crate::{
+    base_committer::base_committer_builder::BaseCommitterBuilder, block::BlockAPI,
+    commit::LeaderStatus, context::Context, dag_state::DagState, storage::mem_store::MemStore,
+    test_dag_parser::parse_dag,
+};
+
+#[tokio::test]
+async fn direct_commit() {
+    telemetry_subscribers::init_for_testing();
+    let context = Arc::new(Context::new_for_test(4).0);
+    let dag_state = Arc::new(RwLock::new(DagState::new(
+        context.clone(),
+        Arc::new(MemStore::new()),
+    )));
+    let committer = BaseCommitterBuilder::new(context.clone(), dag_state.clone()).build();
+
+    // Round 3 is a leader round
+    // D3 is an elected leader for wave 1
+    // Round 4 is a voting round
+    // Round 5 is a decision round (acknowledge)
+    let dag_str = "DAG {
+        Round 0 : { 4 },
+        Round 1 : { },
+        Round 2 : { },
+        Round 3 : { * },
+        Round 4 : { 
+            A -> [D3],
+            B -> [D3],
+            C -> [D3],
+            D -> [], 
+        },
+        Round 5 : { 
+            A -> [A4, B4, C4],
+            B -> [A4, B4, C4],
+            C -> [A4, B4, C4],
+            D -> [], 
+        },
+        }";
+
+    let (_, dag_builder) = parse_dag(dag_str).expect("a DAG should be valid");
+    dag_builder.persist_all_blocks(dag_state.clone());
+
+    let leader_round = committer.leader_round(1);
+    tracing::info!("Leader round at wave 1: {leader_round}");
+    let leader = committer
+        .elect_leader(leader_round)
+        .expect("there should be a leader at wave 1");
+    let leader_status = committer.try_direct_decide(leader);
+    if let LeaderStatus::Commit(_) = leader_status {
+        tracing::info!("Committed: {leader_status}");
+    } else {
+        panic!("Expected a committed leader, got {leader_status}");
+    }
+}
+
+#[tokio::test]
+async fn direct_skip() {
+    telemetry_subscribers::init_for_testing();
+    let context = Arc::new(Context::new_for_test(4).0);
+    let dag_state = Arc::new(RwLock::new(DagState::new(
+        context.clone(),
+        Arc::new(MemStore::new()),
+    )));
+    let committer = BaseCommitterBuilder::new(context.clone(), dag_state.clone()).build();
+
+    // Round 3 is a leader round
+    // D3 is an elected leader for wave 1
+    // Round 4 is a voting round
+    // Round 5 is a decision round (acknowledge)
+    let dag_str = "DAG {
+        Round 0 : { 4 },
+        Round 1 : { },
+        Round 2 : { },
```

Comment:
I would recommend to still try and express the DAG in the most complete/valid way . Although the dag builder allows us to do this (which we might want to revisit) , I believe it would be better to have rounds complete

Reply:
Same comment applies for the rest of the tests.

Reply:
Thank you for a feedback @akichidis 

The reason why I decided to omit those, because those initial rounds are kinda strange: it doesn't matter if I define connections or not, the leader of wave 0 is always `Undecided`. So, if I do:

```
DAG {
        Round 0 : { 4 },
        Round 1 : { * },
        Round 2 : { * },
        Round 3 : { * },
};
```
The leader status appears as an `Undecided` anyway. I'm going to investigate it if it's intentional..

Reply:
I experimented with this a bit and added blocks on the genesis. The `dag_state` doesn't like it:

```
thread 'base_committer::base_committer_declarative_tests::direct_commit' panicked at consensus/core/src/dag_state.rs:294:9:
assertion `left != right` failed: Genesis block should not be accepted into DAG.
```